web/src/protect/organizations/users.rs

```diff
@@ -17,7 +18,45 @@ pub(crate) async fn index(
     request: Request,
     next: Next,
 ) -> impl IntoResponse {
-    match user::find_by_organization(app_state.db_conn_ref(), organization_id).await {
+    check_user_in_organization(
+        &app_state,
+        authenticated_user,
+        organization_id,
+        request,
+        next,
+    )
+    .await
+}
+
+/// Checks that the authenticated user is associated with the organization specified by `organization_id`
+/// Intended to be given to axum::middleware::from_fn_with_state in the router
+pub(crate) async fn create(
+    State(app_state): State<AppState>,
+    AuthenticatedUser(authenticated_user): AuthenticatedUser,
+    Path(organization_id): Path<Id>,
+    request: Request,
+    next: Next,
+) -> impl IntoResponse {
+    check_user_in_organization(
+        &app_state,
+        authenticated_user,
+        organization_id,
+        request,
+        next,
+    )
+    .await
+
+    // TODO: Check that the authenticated user is a coach
```

Comment:
Did you mean to leave this as a TODO? Is it not straightforward to implement this check at this time?

Reply:
I left it yeah a note that we should check eventually. It's not immediately clear to me whether or not this endpoint will be only for coaches in the future until we work out some of the specifics around the user creation workflow. Ex create user -> assign user to coaching relationship later. 
That's why I left it out at the moment. It may be that we decide on separate endpoints for different "flavors" of user creation.

Reply:
Sounds good and reasonable. Would you mind adding your comment to your TODO as helpful context? I'd be happy with that then.

Reply:
Absolutely! Added d0c6dfd71e1f0352288265bd8555fc54c18e3b30